packages/hint-compat-api/src/helpers/compat-html.ts

```diff
@@ -23,18 +23,20 @@ export class CompatHTML extends CompatBase<Events, Event> {
         this.walk((elementFound: ElementFound) => {
             const { element, resource } = elementFound;
             const location = element.getLocation();
+            const codeSnippet = element.outerHTML;
```

Comment:
Also, this can end up including a serialization of most of the page if an element like `<main>` is targeted. Let's add a `utils` helper to do the start-tag-only behavior we've been discussing in #2211. No need to update other hints in this PR, but I'd like to avoid adding new large serializations.

Reply:
@antross done!